dqc_us_rules/dqc_us_0041.py

```diff
@@ -0,0 +1,312 @@
+# (c) Copyright 2016, XBRL US Inc. All rights reserved.
+# See license.md for license information.
+# See PatentNotice.md for patent infringement notice.
+from .util import messages
+
+import json
+import os
+import re
+import time
+
+from arelle.ModelDtsObject import ModelConcept
+from arelle import XbrlConst, ModelXbrl
+from arelle.FileSource import openFileStream, saveFile, openFileSource
+
+
+_CODE_NAME = 'DQC.US.0041'
+_RULE_VERSION = '1.1'
+
+_EARLIEST_GAAP_YEAR = 2014
+
+ugtDocs = (
+    {
+        "year": 2014,
+        "namespace": "http://fasb.org/us-gaap/2014-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2014/elts/us-gaap-doc-2014-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2014/entire/us-gaap-entryPoint-std-2014-01-31.xsd",  # noqa
+     },
+
+    {
+        "year": 2015,
+        "namespace": "http://fasb.org/us-gaap/2015-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/elts/us-gaap-doc-2015-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2015/us-gaap-2015-01-31.zip/us-gaap-2015-01-31/entire/us-gaap-entryPoint-std-2015-01-31.xsd",  # noqa
+     },
+
+    {
+        "year": 2016,
+        "namespace": "http://fasb.org/us-gaap/2016-01-31",
+        "docLB": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/elts/us-gaap-doc-2016-01-31.xml",  # noqa
+        "entryXsd": "http://xbrl.fasb.org/us-gaap/2016/us-gaap-2016-01-31.zip/us-gaap-2016-01-31/entire/us-gaap-entryPoint-std-2016-01-31.xsd",  # noqa
+     }
+)
+
+
+def _make_cache(val, ugt, cntlr, ugt_default_dimensions_json_file):
+    """
+    Creates a new caches for the Taxonomy default dimensions
+
+    :param val: ValidateXbrl to be validated
+    :type val: :class: '~arelle.ValidateXbrl.ValidateXbrl'
+    :param ugt: Taxonomy to check
+    :type ugt: str
+    :param ugt_default_dimensions_json_file: location to save json default
+        dimensions
+    :type ugt_default_dimensions_json_file: str
+    :return: no explicit return, but saves caches for dqc_us_0041
+    :rtype: None
+    """
+    started_at = time.time()
+    ugt_entry_xsd = ugt["entryXsd"]
+    val.usgaapDefaultDimensions = {}
+    prior_validate_disclosure_system = (
+        val.modelXbrl.modelManager.validateDisclosureSystem
+    )
+    val.modelXbrl.modelManager.validateDisclosureSystem = False
+    calculations_instance = (
```

Comment:
This should probably be called something else, right?  Since we aren't doing calcs with it anymore